cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -431,9 +431,13 @@ func tryDelWithPrevResult(driverClient driver.NetworkAPIs, conf *NetConf, k8sArg
 		return false, nil
 	}
 	podVlanID, err := strconv.Atoi(dummyIface.Mac)
-	if err != nil || podVlanID == 0 {
+	if err != nil {
 		return true, errors.Errorf("malformed vlanID in prevResult: %s", dummyIface.Mac)
 	}
+	// If VLAN value is 0, return without handling so that normal delete logic can occur.
```

Comment:
Can we make `0` a constant something like `reservedVlanId` or anything.. and additionally (might be there already) if during add, for whatever reason vlan id is 0 for SGPP , we need to bail out.

Reply:
Yeah, ee already guard against VLAN ID of 0 during add in older releases: https://github.com/aws/amazon-vpc-cni-k8s/blob/release-1.11/cmd/routed-eni-cni-plugin/cni.go#L217 . As for making this a constant, I think that might fit better in the master branch: https://github.com/aws/amazon-vpc-cni-k8s/blob/master/cmd/routed-eni-cni-plugin/cni.go#L448 . I am a bit wary of changing this older release branch more than just to fix this bug

Reply:
Yeah we can change on both since it will be just a const..

Reply:
Added in latest revision